marxs/optics/polarization.py

```diff
@@ -0,0 +1,28 @@
+import numpy as np
+
+from ..math.pluecker import *
+
+
+def polarization_vectors(dir_array, angles):
+	n = len(angles)
+	polarization = np.zeros((n, 4))
+	x = np.array([1., 0., 0.])
+	y = np.array([0., 1., 0.])
+	for i in range(0, n):
+		r = h2e(dir_array[i])
+		r /= np.linalg.norm(r)
+		if not (np.isclose(r[0], 0.) and np.isclose(r[2], 0.)):
+			# polarization relative to positive y at 0              
```

Comment:
Good point! I think I'm going to keep the convention, because I don't know of any easier alternative, but this condition needs to be checked anytime photon directions change (so far only in the mirror.) I'll add a function to the polarization file.